packages/grant-explorer/src/features/common/PassportWidget.tsx

```diff
@@ -0,0 +1,175 @@
+import { useEffect, useState } from "react";
+import { PassportResponse, fetchPassport } from "../api/passport";
+import { useAccount } from "wagmi";
+import { ReactComponent as GitcoinPassportLogo } from "../../assets/passport-logo.svg";
+import { ReactComponent as GitcoinPassportLogoFull } from "../../assets/passport-logo-full.svg";
+import { Dropdown as DropdownIcon } from "common/src/icons/Dropdown";
+
+export enum PassportState {
+  NOT_CONNECTED,
+  INVALID_PASSPORT,
+  SCORE_AVAILABLE,
+  LOADING,
+  ERROR,
+  INVALID_RESPONSE,
+}
+
+export function PassportWidget() {
+  const [, setPassport] = useState<PassportResponse | undefined>();
+  const [, setError] = useState<Response | undefined>();
+  const { address, isConnected } = useAccount();
+
+  const [passportState, setPassportState] = useState<PassportState>(
+    PassportState.LOADING
+  );
+
+  const [passportScore, setPassportScore] = useState<number>();
+  const [textColor, setTextColor] = useState<string>("gray");
+  const [donationImpact, setDonationImpact] = useState<number>(0);
+  const [open, setOpen] = useState(false);
+
+  function handleClick() {
+    if (
+      passportState === PassportState.SCORE_AVAILABLE ||
+      passportState === PassportState.INVALID_PASSPORT
+    ) {
+      setOpen(!open);
+    }
+  }
+
+  useEffect(() => {
+    setPassportState(PassportState.LOADING);
+
+    const PASSPORT_COMMUNITY_ID =
+      process.env.REACT_APP_PASSPORT_API_COMMUNITY_ID;
+
+    if (isConnected && address && PASSPORT_COMMUNITY_ID) {
+      const callFetchPassport = async () => {
+        const res = await fetchPassport(address, PASSPORT_COMMUNITY_ID);
+        if (res.ok) {
+          const scoreResponse = await res.json();
+
+          if (scoreResponse.status === "PROCESSING") {
+            console.log("processing, calling again in 3000 ms");
+            setTimeout(async () => {
+              await callFetchPassport();
+            }, 3000);
+            return;
+          }
+
+          if (scoreResponse.status === "ERROR") {
+            // due to error at passport end
+            setPassportState(PassportState.ERROR);
+            return;
+          }
+
+          setPassport(scoreResponse);
```

Comment:
there's already a usePassport hook for this - we should reuse it.

Reply:
@vacekj  Yes, agree. I didn't reuse it initially because, the hook was just returning matching available or not state but not the actual score, which we need in the widget.

Will update the hook itself now as we've decided to remove the passport banner & updating the passport logic across all our apps.